pybossa/task_creator_helper.py

```diff
@@ -40,23 +41,21 @@ def s3_conn_type():
     return current_app.config.get('S3_CONN_TYPE')
 
 
-def get_task_expiration(current_expiration):
+def get_task_expiration(expiration, create_time):
```

Comment:
You can have create_time=None and 
```
if create_time is None:
  create_time = make_timestamp()
```
This can avoid passing make_timestamp() in multiple places.
BTW, I know it is not introduced in the PR, but make_timestamp() should not reside in model class, and classes other than 'repository' classes, should not call functions in model classes directly. Perhaps a better place for make_timestamp is the util module.

Reply:
`create_time` null is an invalid scenario as tasks shouldn't be created with null value. Also, with the code snippet you mentioned, there's possibility of having expiration beyond max allowed expiration which we want to avoid. `make_timestamp` is in model class originally and called by many other model classes.